model/market.ts

```diff
@@ -357,3 +359,76 @@ export const calculateDepositApy = (
     proceeds: totalDeposit - initialDeposit,
   }
 }
+
+export const calculateBorrowApr = (
+  substitute: Currency,
+  markets: Market[],
+  initialBorrow: bigint,
+  maxAmountExcludingFee: bigint,
+  currentTimestamp: number,
+): {
+  interest: bigint
+  apr: number
+  totalBorrow: bigint
+  available: bigint
+} => {
+  if (
+    markets.some(
+      (market) =>
+        !isAddressEqual(
+          market.quoteToken.address,
+          substitute.address as `0x${string}`,
+        ),
+    )
+  ) {
+    new Error('Substitute token is not supported')
+  }
+
+  const availableCoupons = min(
+    ...markets.map((market) => market.totalAsksInBaseAfterFees()),
+  )
+  const interestAvailableCoupons = min(
```

Comment:
Using `min` here would result in `0n` most of the time.

Reply:
https://github.com/clober-dex/coupon.finance/pull/74/commits/cfd87c95a76e89e7926ea7a116def42fcee37ef0

Oops, that was a significant issue. Thanks for catching it! @onemonster